app/models/order.rb

```diff
@@ -1,35 +1,40 @@
 class Order < ActiveRecord::Base
+  attr_accessor :confirmed_payment
+
+  after_initialize do |order|
+    order.confirmed_payment = false
+  end
+
   # DB relationships
   has_many :order_items, dependent: :destroy
   # should destroy all of the associated OrderItems if an Order is destroyed.
-  # we can use this as part of the cleaning task we set up to kill any pending
-  # orders inactive for whatever time we set.
-  # (20-30 minutes? a day? anything inactive for over 2hrs, but only run task once a day?)
+  # TODO: but do we want to destroy Orders?
   has_many :products, through: :order_items
 
 
   # validations helper regex
   # email regex from: http://rails-3-2.railstutorial.org/book/modeling_users#code-validates_format_of_email
   VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-]+(\.[a-z\d\-]+)*\.[a-z]+\z/i
+  VALID_BUYER_CARD_SHORT_REGEX = /\A\d{4}\z/
 
   # data validations
   validates :status, presence: true, inclusion: { in: %w(pending paid complete canceled),
     message: "%{value} is not a valid status" }
 
-  validates_presence_of :buyer_email, unless: :pending?
-  validates_format_of :buyer_email, with: VALID_EMAIL_REGEX, unless: :pending?
-
-  validates_presence_of :buyer_name, unless: :pending?
-  validates_presence_of :buyer_address, unless: :pending?
-  # TODO: validate address or name somehow?
+  with_options unless: :pending? do
+    validates_presence_of :buyer_email
+    validates_format_of :buyer_email, with: VALID_EMAIL_REGEX
 
-  validates_presence_of :buyer_card_short, unless: :pending?
-  validates_numericality_of :buyer_card_short, only_integer: true, greater_than: 999, less_than: 10_000, unless: :pending?
-  # FIXME: these will not handle for cards that end in: 0812, 0002, 0404, etc.
+    # TODO: should we validate buyer name & address any futher?
```

Comment:
I don't think so. I've heard some pretty compelling arguments that it's a really bad idea to try to "standardize" someone's name or address.